### PR TITLE
support for embedded youtube and vimeo videos

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/1085

### CODE REVIEW

File: src/components/Proposal/ProposalDetailsPage.tsx [hunk 1]
```diff
@@ -97,6 +97,28 @@ class ProposalDetailsPage extends React.Component<IProps, IState> {
...
+    }
+  }
+
+  private getVimeoIdFromUrl = (url: string): string => {
+    const videoid = url.match(/^.*(?:vimeo.com)\/(?:channels\/|channels\/\w+\/|groups\/[^/]*\/videos\/|album\/\d+\/video\/|video\/|)(\d+)(?:$|\/|\?)/);
+    if(videoid && (videoid.length >= 2)) {
+      return videoid[1];
+    } else { 
+      // eslint-disable-next-line no-console
+      console.error("The youtube url is not valid.");
```
Comment (tibetsprague) [inline src/components/Proposal/ProposalDetailsPage.tsx:117]: Vimeo

File: src/components/Proposal/ProposalDetailsPage.tsx [hunk 2]
```diff
@@ -198,6 +220,25 @@ class ProposalDetailsPage extends React.Component<IProps, IState> {
...
+                    const url = new URL(props.href);
+                    const videoId = this.parseYouTubeVideoIdFromUri(props.href);
+                    if (videoId) {
+                      const start = url.searchParams.get("t") || "0";
+
+                      return <iframe className={css.embeddedVideo} frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen
+                        src={`${url.protocol}//www.youtube-nocookie.com/embed/${videoId}?start=${start}`}>
+                      </iframe>;
+                    }
+                    else {
```
Comment (tibetsprague) [inline src/components/Proposal/ProposalDetailsPage.tsx:233]: usually we put this on the same line as the } above

File: src/components/Proposal/ProposalDetailsPage.tsx [hunk 3]
```diff
@@ -97,6 +97,28 @@ class ProposalDetailsPage extends React.Component<IProps, IState> {
     this.setState({ showVotersModal: false });
   }
 
+  private parseYouTubeVideoIdFromUri = (url: string): string => {
+    const videoid = url.match(/(\/|%3D|v=)([0-9A-z-_]{11})([%#?&]|$)/);
+    if(videoid && (videoid.length >= 3)) {
```
Comment (tibetsprague) [inline src/components/Proposal/ProposalDetailsPage.tsx:102]: space after the if

File: src/components/Proposal/ProposalDetailsPage.tsx [hunk 4]
```diff
@@ -97,6 +97,28 @@ class ProposalDetailsPage extends React.Component<IProps, IState> {
...
+    } else { 
+      // eslint-disable-next-line no-console
+      console.error("The youtube url is not valid.");
+      return null;
+    }
+  }
+
+  private getVimeoIdFromUrl = (url: string): string => {
+    const videoid = url.match(/^.*(?:vimeo.com)\/(?:channels\/|channels\/\w+\/|groups\/[^/]*\/videos\/|album\/\d+\/video\/|video\/|)(\d+)(?:$|\/|\?)/);
+    if(videoid && (videoid.length >= 2)) {
```
Comment (tibetsprague) [inline src/components/Proposal/ProposalDetailsPage.tsx:113]: space after the if

File: src/components/Proposal/ProposalDetailsPage.tsx (hunk 2 above)
Comment (dkent600) [reply src/components/Proposal/ProposalDetailsPage.tsx:237]: fixed

File: src/components/Proposal/ProposalDetailsPage.tsx (hunk 1 above)
Comment (dkent600) [reply src/components/Proposal/ProposalDetailsPage.tsx]: fixed, and note I also improved the error handling logic

File: src/components/Proposal/ProposalDetailsPage.tsx (hunk 2 above)
Comment (dkent600) [reply src/components/Proposal/ProposalDetailsPage.tsx:237]: fixed

File: src/components/Proposal/ProposalDetailsPage.tsx (hunk 3 above)
Comment (dkent600) [reply src/components/Proposal/ProposalDetailsPage.tsx]: hmm, seems like our linter ought to catch that.

fixed, and note I also improved the error handling logic

File: src/components/Proposal/ProposalDetailsPage.tsx (hunk 4 above)
Comment (dkent600) [reply src/components/Proposal/ProposalDetailsPage.tsx]: hmm, seems like our linter ought to catch that.

fixed, and note I also improved the error handling logic

**Pull request merged**